kin-devplatform-sdk/src/main/java/kin/devplatform/accountmanager/AccountManagerImpl.java

```diff
@@ -229,8 +234,44 @@ public void onError(Exception e) {
 	}
 
 	@Override
-	public void switchAccount(final int accountIndex, @NonNull final KinCallback<Boolean> callback) {
+	public void switchAccount(final int accountIndex, @NonNull final KinCallback<Boolean> callback,
+		@NonNull final IMigrationManagerCallbacks migrationManagerCallbacks) {
 		Logger.log(new Log().withTag(TAG).put("switchAccount", "start"));
+		startMigration(accountIndex, callback, migrationManagerCallbacks);
+	}
+
+	private void startMigration(final int accountIndex, final KinCallback<Boolean> callback,
+		final IMigrationManagerCallbacks migrationManagerCallbacks) {
+		Logger.log(new Log().withTag(TAG).put("startMigration", "start"));
+		try {
+			migrationManager.start(new IMigrationManagerCallbacks() {
+				@Override
+				public void onMigrationStart() {
+					Logger.log(new Log().priority(Log.DEBUG).withTag(TAG).text("onMigrationStart"));
+					migrationManagerCallbacks.onMigrationStart();
+				}
+
+				@Override
+				public void onReady(IKinClient kinClient) {
+					Logger.log(new Log().priority(Log.DEBUG).withTag(TAG).text("onReady"));
+					blockchainSource.updateKinClientAndAccount(kinClient, accountIndex);
```

Comment:
you must update blockchain source only after a successful update to wallet address (like its done in `updateWalletAddress.onResponse) o.w. we can end up in a limbo state

Reply:
Fixed